src/components/MainContent/MainContent.tsx

```diff
@@ -0,0 +1,70 @@
+import { useContext, useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { PostsList } from '../PostsList';
+import { ModalUserContext } from '../ModalUserContext';
+import { Post } from '../../types/Post';
+import { getPostsByUserId } from '../../api/api';
+
+export const MainContent = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoad, setIsLoad] = useState(false);
+  const { modalUser } = useContext(ModalUserContext);
+
+  const showLoad = isLoad && !isError && !posts.length;
+  const showError = !isLoad && isError && !posts.length;
+  const showPostList = !isLoad && !isError && !!posts.length;
+  const showNoPostsYet = !isLoad && !isError && !posts.length;
```

Comment:
Functions are usually named from the verb, and you have variables.
In addition, you do similar checks in another place, so it can be transferred to a separate helper, where data can be passed as an argument, compared in the middle of the helper and return a Boolean result, and reused in two places.
<img width="453" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-posts/assets/54860641/350e417c-87e0-4ea9-861a-892d56aca357">